Source/EasyNetQ/Consumer/HandlerCollection.cs

```diff
@@ -71,13 +67,15 @@ public HandlerCollection(IEasyNetQLogger logger)
             return (message, info) => Task.Factory.StartNew(() => { });
         }
 
-        public dynamic GetHandler(Type messageType)
+        public Func<IMessage<object>, MessageReceivedInfo, Task> GetHandler(Type messageType)
         {
             Preconditions.CheckNotNull(messageType, "messageType");
-
+            
             var getHandlerGenericMethod = GetType().GetMethod("GetHandler", new Type[0]).MakeGenericMethod(messageType);
```

Comment:
Is there a reason here to use the overload of `GetMethod(string, Type[])` here ? I would guess the base method that only takes a string should work fine.